intake/source/derived.py

```diff
@@ -268,3 +268,68 @@ def __init__(self, columns, **kwargs):
 
     def pick_columns(self, df):
         return df[self._params["columns"]]
+
+
+class DataFramePipeline(DataFrameTransform):
+    """Apply a sequence of transformations over a DataFrame
+
+    The sequence of steps is defined as a list-of-dicts under
+    the key "steps". Each step requires a "method" key and
+    can optionally take "kwargs".
+
+    A special method called 'cols' enables column selection
+    and takes the kwarg 'columns'.
+
+    If a method cannot be found as an attribute of the output
+    of the previous step, the pipeline will attempt to import
+    the method and apply it.
+
+    Here's an example of performing a groupby, selecting two
+    columns and computing the mean.
+
+    by_origin:
+      driver: intake.source.derived.DataFramePipeline
+      args:
+        targets:
+          - auto
+        steps:
+          - method: groupby
+            kwargs:
+              by: origin
+          - method: cols
+            kwargs:
+              columns:
+                - hp
+                - mpg
+          - method: mean
```

Comment:
Is it OK if the output of the source is not a dataframe? I think yes.

Reply:
Sure!